### ISSUE TITLE
Make site mobile-friendly

### ISSUE THREAD

**Issue by imphil** [librecores/librecores-web#57]
Currently especially the homepage (https://librecores.org/) doesn't look good on mobile devices.
1. Figure out how we can reduce the content to be mobile-friendly. Especially the pictures (engage/code/share/reuse/learn) need to be reduced in size or left out completely.
2. Modify the HTML/CSS code using the Bootstrap provided CSS classes (http://getbootstrap.com/css/) to realize the idea.
3. Create a patch for the librecores-web repository to incorporate the changes into the site.

**Comment by eliaskousk** [comment]
I think the menu is the main reason for the page looking bad on mobile, but then again I'm no frontend expert! Aren't we using bootstrap already? Maybe its easy to make the menu more compatible with bootstrap.

I agree that we should make it reasonably usable on mobile but unless a pure frontend developer joins and helps us with this we won't be able to make a huge improvement (pessimist mode: on!).

For projects that I made myself and didn't have an experienced frontend developer tackling these issues I always used the [Mopa Bootstrap](https://github.com/phiamo/MopaBootstrapBundle) bundle which wraps bootstrap with Symfony in an easy to use way and gives you a lot of helpers for everything that bootstrap supports inside Symfony (e.g menu generation). I don't how this differs from the work you've done here, importing bootstrap manually, maybe there are essentially the same.

**Comment by imphil** [comment]
This bug has been marked as a "good first bug". If you're interested in working on it, have a look at the developer documentation at http://librecores-web.readthedocs.io/en/latest/ and comment on this bug or contact the listed mentor for more details. The mentor will be available to help you complete this task.

Mentor: @imphil
Required Skills: HTML, CSS

**Comment by imphil** [comment]
@berndca, I like the demo at http://librecores.netlify.com/. Very nice work, I especially like the reduced icons.

A couple of minor comments:
- maybe a bit of margin/padding around the logo (top right) would be nice for small screens (<768px)
- the search button below the search bar looks still not really good in my eyes. At least we should add a bit of space between the search bar and the button.
- when 768px wide, the new icons for share/... are positioned not centered, but more on the left. This isn't that visible on smaller screens, but at this width it looks a bit strange.
- The space between the boxes titled "Getting started with your LibreCore" adn "Feel the pulse" is larger than the space between the other boxes.

**Comment by wallento** [comment]
very nice, @berndca.

Another minor comment: The upper part doesn't seem centered.

**Comment by berndca** [comment]
Thanks for the feedback. The main point of the first demo was to see if 
the fa icon approach for small screens would be acceptable.

I have pushed an updated version. Please review and comment.

The issue with the search submit button was another overflow. I have 
change the search input to an "input-group-btn" to avoid overflow.
I would not consider this solution perfect. However I would want to get 
a better understanding of the sites information architecture
before spending more time on this detail.

Stefan, could you please be more specific and refer to some text or 
other landmarks in the area of concern.

Thanks, Bernd

**Comment by imphil** [comment]
Looks great, no further comments on my side!  Thanks a lot! Let me know if you need help preparing a PR for this.

**Comment by wallento** [comment]
Hi, thanks!

For me the "We drive Free and Open Digital Hardware." is not centered but seems leaning to the right.

**Comment by berndca** [comment]
@wallento Good catch. I uploaded a new version.

**Comment by berndca** [comment]
@imphil Due to the fact that CSS and template changes impact more than one page I might want to take a look at the other pages first to make sure I don't break anything. Perhaps it would be a good idea to include updates to all pages in this PR?

**Comment by imphil** [comment]
sounds good to me.

**Comment by berndca** [comment]
@imphil I have quite a number of changes to librecores.css. I have basically switched to the version served by the development server, because it's cleaner. (There a semicolon required even after the last statement in a block.)

I noticed a couple of open PR's. PR #92 also has changes for librecores.css. Any chance that this could be merged soon? There will also be changes to other templates since the grid breakpoint change requires switching from col-md-* to col-sm-* classes. 

Here is a list of template files I already that they need to be changed:
- site/src/Librecores/SiteBundle/Resources/views/Default/home.html.twig
- site/src/Librecores/ProjectRepoBundle/Resources/views/Project/view.html.twig
- site/src/Librecores/ProjectRepoBundle/Resources/views/Project/settings.html.twig:
- site/src/Librecores/ProjectRepoBundle/Resources/views/Default/project_search.html.twig
- site/src/Librecores/ProjectRepoBundle/Resources/views/Organization/settings.html.twig
- site/src/Librecores/PlanetBundle/Resources/views/Default/index.html.twig

Is there am easy way to get all URL's to make sure I'm not missing any pages?

**Comment by imphil** [comment]
@berndca 
- PR #92 is merged now, the other one requires more work. 
- if you search for *.html.twig you should find all HTML templates. Your list looks about right. I don't know of an easy way to get all URLs, you essentially would need to look at the routing.xml and the associated PHP functions which template is rendered. But from your list, that's it:

    site/src/Librecores/SiteBundle/Resources/views/Default/home.html.twig

homepage 

    site/src/Librecores/ProjectRepoBundle/Resources/views/Project/view.html.twig

view project. Create a project and have a look at the librecores.devel/USERNAME/PROJECTNAME page.

    site/src/Librecores/ProjectRepoBundle/Resources/views/Project/settings.html.twig:


view project. Create a project and have a look at the librecores.devel/USERNAME/PROJECTNAME/settings page.

    site/src/Librecores/ProjectRepoBundle/Resources/views/Default/project_search.html.twig


librecores.devel/search

    site/src/Librecores/ProjectRepoBundle/Resources/views/Organization/settings.html.twig

create a new organization, then look at librecores.devel/ORGANIZATION/settings

    site/src/Librecores/PlanetBundle/Resources/views/Default/index.html.twig

the planet home page (librecores.devel/planet)

**Comment by berndca** [comment]
@imphil I could use some help with from template for the search form in the home page.

So far I got: 

```
<form name="search-form" method="get" action="{{ path('librecores_project_repo_project_search') }}">
	<div class="input-group">
		<input type="text" class="form-control" placeholder="What IP is missing in your project?">
		<span class="input-group-btn">
	  <button class="btn btn-danger" type="submit">
		<span class="glyphicon glyphicon-search" aria-hidden="true"></span>
		<span class="hidden-xs">Find IP Project</span></button>
		</span>
	</div>
</form>
```

I noticed that the original template uses form_start, form_widget etc. Do I have to use these functions or would the above code be acceptable?

Thanks, Bernd

**Comment by imphil** [comment]
@berndca the form_* symfony tags make sure that the default styling is added to them. For the search box on the home page (and since it's just a very simple GET form anyways) going with plain HTML is fine (at least I don't see any problems right now) and makes styling easier. Go for it!

**Comment by berndca** [comment]
Sorry guys the new navbar item (CI) causes overflow again. Please do seriously consider saving some characters in the navbar. We were pretty much at the border before adding CI. Therefore the total number of characters in the navbar should not exceed what we had before. Can you make this work?

Please find a live demo with the extended navbar at http://librecores.netlify.com/.

**Comment by imphil** [comment]
I also find the navbar a bit overcrowded -- we don't need to have all content in the top navbar.

@wallento, @berndca how about this reduced version?

- Search Projects
- Learning Center (Documentation, link to https://www.librecores.org/static/docs)
- Community Hub (Replaces Planet LibreCores, link to https://www.librecores.org/static/community-hub)
- About & Roadmap (leave that one in for now, I think the "Roadmap" part can go if we have a bit more content)

**Comment by berndca** [comment]
Would it help to group the items and use dropdowns?

**Comment by wallento** [comment]
My thoughts:

- Instead of "Search Projects" we probably will have "Browse" in long term. I like how an some websites they have a magnifier symbol and when you click it expands to a search bar.
- "Getting Started" should stay
- "Planet Librecores" -> "Planet"
- "Continuous Integragtion" -> "LibreCores CI"
- "About & Roadmap" -> "About" (the roadmap was from a previous marketing plan of me)

**Comment by wallento** [comment]
That's probably the least impact and (rename only) and we can later think of dropdowns (which I also like). If you both agree I will create a PR

**Comment by imphil** [comment]
- Search vs. Browse: I agree with the long-term plan, but for now there's nothing other than search :)
- I don't like the "Getting Started" one -- getting started with what exactly? That's why something in the line of "Learning Center" would work out better I think.
- Planet is just one thing, we also have quite some mailing lists, irc channels etc. we should link the more general category on the toplevel. Then move the Planet into a sub-page or a drop-down (I think that would be nice option)
- let's keep CI then.

**Comment by wallento** [comment]
Understood.
- Searching is also not very fascinating, but okay, then "Search"?
- "Learning Center" sounds a bit academic to me, so I would compromise on "Learn".
- "Community" instead of "Community Hub"?

Of course I start bikeshedding now, make a final call and I can fix it immediately.

**Comment by berndca** [comment]
I have no strong opinion either way. Just a few observations. The "Search Projects" link leads to the Search page which implements search for projects AND users. Perhaps we could change it to just Search?



The dropdowns would be easy to implement.

**Comment by wallento** [comment]
Please amend as it makes sense, @imphil 

Thanks all!

**Comment by imphil** [comment]
> I have no strong opinion either way. Just a few observations. The "Search Projects" link leads to the Search page which implements search for projects AND users. Perhaps we could change it to just Search?

Hm. Probably some reference to the fact that LibreCores hosts Projects would be nice there? I agree that "Search Projects" is technically not correct, but maybe worth to have "Projects" in there? Or go directly with the result, and not the action, making that "Find Projects"? So many options :-)

So let's go with:
- Search/Search Projects. Or go with "Find Projects"?
- Learn
- Community
  - Blog Planet
  - Community Hub/Resources (no strong opinion here)
- LibreCores CI
- About

**Comment by wallento** [comment]
Yes, that looks good to me. I found that in the PR I even missed LibreCores CI as CI. I would amend, but how would drop down work?

**Comment by imphil** [comment]
@berndca how about you add the new header directly in your PR? Since you probably know best how to implement the dropdown as well.

**Comment by berndca** [comment]
Have a look: http://librecores.netlify.com/short-1/ (**new** link)

**Comment by berndca** [comment]
@imphil Does this work for you? Let me know if you want me to change anything.

**Comment by imphil** [comment]
@berndca looks good! A small (and I think last) bit on the jumbotron.

In Desktop and Mobile:  could you increase the space between the "We drive Free and Open Digital Hardware." and the search bar? It looks a bit close together.

With FF on my cellphone it looks like the picture below. Could we reduce the size of the font a bit? It looks rather large in comparison to the search bar below it.

![grafik](https://cloud.githubusercontent.com/assets/1467123/22386254/943fe94e-e4d6-11e6-940e-ae7096a56fff.png)

**Comment by berndca** [comment]
I have updated http://librecores.netlify.com/short-1/. Please review and comment.

**Comment by imphil** [comment]
Looks good!

**Comment by imphil** [comment]
Fixed in #100
